src/async_impl/decoder.rs

```diff
@@ -68,17 +27,32 @@ mod imp {
         inner: Inner,
     }
 
+    enum DecoderType {
+        #[cfg(feature = "gzip")]
+        Gzip,
+        #[cfg(feature = "brotli")]
+        Brotli,
+    }
+
     enum Inner {
         /// A `PlainText` decoder just returns the response content as is.
         PlainText(super::super::body::ImplStream),
+
         /// A `Gzip` decoder will uncompress the gzipped response content before returning it.
+        #[cfg(feature = "gzip")]
         Gzip(GzipDecoder<Peekable<IoStream>>),
+
+        /// A `Brotli` decoder will uncompress the brotlied response content before returning it.
+        #[cfg(feature = "brotli")]
+        Brotli(BrotliDecoder<Peekable<IoStream>>),
+
         /// A decoder that doesn't have a value yet.
+        #[allow(dead_code)]
```

Comment:
Does this sometimes trigger dead code?

Reply:
If gzip and brotli are not enabled, dead code will be triggered

Reply:
Oh I see. So then, this variant could have `#[cfg(any(feature = "brotli", feature = "gzip"))]`, and then when neither is enabled, the extra size isn't used.